pkg/agent/wireguard/client_test.go

```diff
@@ -376,5 +382,85 @@ func Test_DeletePeer(t *testing.T) {
 		_, ok := client.peerPublicKeyByNodeName.Load("fake-node-1")
 		assert.False(t, ok)
 	})
+}
+
+func Test_New(t *testing.T) {
+	client := fake.NewSimpleClientset()
+	_, err := New(client, &config.NodeConfig{Name: "test"}, &config.WireGuardConfig{})
+	require.NoError(t, err)
+}
+
+type testError struct {
+	msg string
+}
+
+func (te testError) Is(target error) bool { return target == unix.EOPNOTSUPP }
```

Comment:
Is this necessary? Can we just assign `unix.EOPNOTSUPP` to the `linkAddErr` below?

Reply:
No, we are using `errors.Is(err, unix.EOPNOTSUPP)` to check the type, and `unix.EOPNOTSUPP` doesn't support `errors.Is` method. `unix.EOPNOTSUPP` is already wrapped by the error returned by netlink.LinkAdd.

Reply:
`func errors.Is(err error, target error) bool` accept two errors as params. BTW I think `unix.EOPNOTSUPP` also implements the interface `Is(target error) bool`.

Reply:
Hmm, could you show me where does `unix.EOPNOTSUPP` implement `Is(target error)`?
I can only find the line like `EOPNOTSUPP      = syscall.Errno(0x40)` in `golang.org/x/sys/unix`.

Reply:
Seems syscall.Errno added [this interface]( https://pkg.go.dev/syscall#Errno.Is) since 1.13.

Reply:
You are right, I didn't check `syscall.Errno`, removed the defined `testError`, thanks.